core/src/com/unciv/logic/automation/UnitAutomation.kt

```diff
@@ -157,9 +157,16 @@ object UnitAutomation {
             // For now its a simple option to allow AI to win a science victory again
             if (unit.hasUnique(UniqueType.AddInCapital))
                 return SpecificUnitAutomation.automateAddInCapital(unit)
-            
-            if (unit.hasUnique("Bonus for units in 2 tile radius 15%"))
-                return SpecificUnitAutomation.automateGreatGeneral(unit)
+
+            //todo this now supports "Great General"-like mod units not combining 'aura' and citadel
+            // abilities, but not additional capabilities if automation finds no use for those two
+            if (unit.hasGreatGeneralUnique && SpecificUnitAutomation.automateGreatGeneral(unit))
+                return
+            val isCitadelPlacer = unit.hasUnique(UniqueType.TakeOverTilesAroundWhenBuilt)
```

Comment:
Wouldn't the improvement have that unique and not the unit?

Reply:
Good one. You're right, I don't remember testing how AI places citadels, focusing on modded boni of varying radii...

This code was a lot less flexible before, a modded unit able to place Citadels without the Strength bonus would never be usable to the AI, thus my changes - will need to re-think how to effectively cache that two-step test.